File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/Row.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.Component;
import io.intino.alexandria.ui.displays.notifiers.RowNotifier;

public class Row<DN extends RowNotifier, Type, B extends Box> extends Component<DN, B> {
    private Type item;

    public Row(B box) {
        super(box);
    }

    public Type item() {
        return this.item;
    }

    public Row item(Type item) {
        this.item = item;
        return this;
    }
}

File: web-ui/src/displays/components/Row.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractRow from "../../../gen/displays/components/AbstractRow";
import RowNotifier from "../../../gen/displays/notifiers/RowNotifier";
import RowRequester from "../../../gen/displays/requesters/RowRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';

const styles = theme => ({});

class Row extends AbstractRow {

	constructor(props) {
		super(props);
		this.notifier = new RowNotifier(this);
		this.requester = new RowRequester(this);
	};


}

export default withStyles(styles, { withTheme: true })(Row);
DisplayFactory.register("Row", withStyles(styles, { withTheme: true })(Row));